frontend/src/components/elements/Video/Video.tsx

```diff
@@ -72,6 +72,21 @@ export default function Video({ element, width }: VideoProps): ReactElement {
        NOTE: This part assumes the URL is already an "embed" link.
     */
   if (type === VideoProto.Type.YOUTUBE_IFRAME) {
+    /* height of IFrame element needs to be set to avoid scrolling issue: https://github.com/streamlit/streamlit/issues/5069
+       initially IFrame element has height properly setup, but after YouTube content is loaded React renders Video with width 0
+       to fix this we save the original width to localStorage key iframe-width-${YouTubeURL},
+       when React tries to render Video component with width 0, we read the initialWidth from localStorage
+       and properly setup component's width and height
+     */
+    const localStorageIFrameWidthKey = `iframe-width-${url}`
+    let initialWidth = window.localStorage.getItem(localStorageIFrameWidthKey)
+    if (width !== 0) {
+      window.localStorage.setItem(localStorageIFrameWidthKey, width.toString())
```

Comment:
I think it would be better here to use `useState` hook to keep a non-zero width in the state. I think `localStorage` is mainly for things we want to persist across sessions.

Reply:
@LukasMasuch this issue starts to be really tricky and interesting, I've just changed the `Video` component to use state instead of localStorage, and I get `initial width` set to 0, despite it being set with `setInitialWidth` 🤔

Reply:
![image](https://user-images.githubusercontent.com/78742618/184215045-dc0383b6-453b-4ece-a3bf-187f25777c7a.png)